shell/platform/embedder/embedder.h

```diff
@@ -1966,6 +1952,143 @@ FlutterEngineResult FlutterEnginePostCallbackOnAllNativeThreads(
     FlutterNativeThreadCallback callback,
     void* user_data);
 
+//------------------------------------------------------------------------------
+/// @brief    Posts updates corresponding to display changes to a running engine
+///           instance.
+///
+/// @param[in] update_type      The type of update pushed to the engine.
+/// @param[in] displays         The displays affected by this update.
+/// @param[in] display_count    Size of the displays array, must be at least 1.
+///
+/// @return the result of the call made to the engine.
+///
+FLUTTER_EXPORT
+FlutterEngineResult FlutterEngineNotifyDisplayUpdate(
+    FLUTTER_API_SYMBOL(FlutterEngine) engine,
+    FlutterEngineDisplaysUpdateType update_type,
+    const FlutterEngineDisplay* displays,
+    size_t display_count);
+
+#endif  // !FLUTTER_ENGINE_NO_PROTOTYPES
+
+/// Function-pointer-based versions of the APIs above.
+typedef struct {
```

Comment:
I'd suggest two minor updates to the API.

1. To make it easier for callers to write their own macros that conditionally use either the proc addresses directly or refer to a proc table, we should ignore the style guide on the naming of struct members and name fields the same as the names of the functions (perhaps without the `FlutterEngine` prefix as that is common). Then, embedders can write their own macros using simple token concatenation. See the pattern followed when using the Vulkan API.
2. Let's typedef the functions for clarity. Though we can do this by sticking to the style guide, lets makes sure that token concatenation gives valid identifiers for users.

After the suggestions, the field for `FlutterEngineInitialize` would be:
```
FlutterEngineInitializeFnPtr Initialize;
```

Reply:
I'd not done the typedef since it increases our maintenance overhead, but I guess the tradeoff is that it makes some users have to define it themselves. I'll update that.

Reply:
All done as suggested.

Note that `FlutterPlatformMessageCreateResponseHandle` and `FlutterPlatformMessageReleaseResponseHandle` are incorrectly named, starting with `Flutter` but not `FlutterEngine`. We should strongly consider adding correctly-named variants and deprecating the old ones. I've named the function pointer types in this version under the assumption that we will do that (i.e., currently those function pointer type names are inconsistent with the actual function names).